File: Retos/Reto #35 - PRIMEROS PASOS [Fácil]/lua/Hugovrc.lua
-- Hola Mundo
print("Hola, Mundo!")

-- Variable tipo String
nombre = "Solobino"

-- Variable numerica entera
edad = 9

-- Variable numerica decimal
peso = 13.5

-- variable booleana
perro = true

-- Uso de if, else if, y else
if edad <= 2 then
    print("Cachorro")
elseif edad > 2 and edad < 10 then
    print("perro adulto")
else 
    print("perro viejito")
end

-- creacion de un Array, en realidad les llaman tablas en lua que parecen diccionarios
mi_array = {"lua", "python", "C#", "Java"}

-- La sentencia for
for llave, valor in ipairs(mi_array)
do 
    print(valor)
end

-- La sentencia While
a = 1
print(#mi_array)
while(a <= #mi_array)
do
    print(mi_array[a])
    a = a + 1
end

-- Funcion sin parametros
function hola_mundo()
    print("Hola, Mundo!")
end

-- Funcion con parametros
function suma(numero1, numero2)
    print(numero1 + numero2)
end

-- Funcion sin parametros y retorno
function saludo()
    return "Hola"
end

-- Funcion con parametros y retorno
function multiplicacion(num1, num2)
    return num1 * num2
end

-- implementacion de una Clase (en realidad se simula por  que no existe como tal)
-- Definicion de la Clase
Auto = {}
Auto.__index = Auto

-- Constructor de la Clase
function Auto.new(marca, anio)
    local self = setmetatable({}, Auto)
    self.marca = marca
    self.anio = anio
    return self
end

File: Retos/Reto #35 - PRIMEROS PASOS [Fácil]/python/Hugovrc.py
# Hola Mundo
print("Hola mundo!")

# Variable tipo String
nombre= "Solobino"

# Variable numerica entera
edad = 2

# Variable numerica decimal
peso = 13.5

# variable booleana
perro = True

# Constante
RAZA = "pitbull"

# Uso de if, else if, y else
if edad <=2:
    print("cachorro")
elif edad > 2 and edad < 10:
    print("perro Adulto")
else:
    print("perro Viejito")

# creacion de una Lista
marcas_autos: list = ["Ford","Nissan","Toyota","Mazda","SEAT"]

# creacion de una Tupla
razas_perro: tuple = ("Beagle","pitbull","Poodle")

# creacion de un Set
paises: set = {"Mexico", "España", "Colombia", "Brazil", "Argentina"}

# creacion de un Diccionario
calificaciones: dict = {"Gerardo" : 90, "Luis" : 85, "Alejandro" : 100, "Andres" : 98}

# La sentencia for
for raza in razas_perro:
    print(raza)

# La sentencia while
indice = 0
while indice < len(marcas_autos):
    print(marcas_autos[indice])
    indice += 1

# Funcion sin parametros
def imprimir_hola_mundo():
    print("Hola mundo")

# Funcion con parametros
def imprimir_nombre_edad(nombre: str, edad: int):
    print(f"Nombre: {nombre}, Edad: {edad}")

# Funcion con parametros con retorno
def imprime_nombre(nombre: str) -> str:
    return nombre

# Creacion de una Clase
class Vehiculo():
    
    def __init__(self, color, marca, ruedas):
        self.color = color
        self.marca = marca
        self.ruedas = ruedas

# Control de excepciones
buscar_indice = 5
try:
    print(marcas_autos[buscar_indice])
except IndexError:
    print("El indice que buscas no existe")










